test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,230 @@
+package specs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ghodss/yaml"
+	apiappsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	v20180930preview "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp"
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Scale Up/Down E2E tests [ScaleUpDown] [Real] [LongRunning] [EveryPR]", func() {
+	var (
+		azurecli *azure.Client
+		occli    *openshift.Client
+		manifest = flag.String("manifest", "../../_data/manifest.yaml", "Path to the manifest to send to the RP")
+	)
+
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewCustomerAdminClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Scaling out the compute scale set to 2 nodes")
+		By("Parsing the external manifest")
+		external, err := parseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By(fmt.Sprintf("Updating the size of the compute scale set to %d in the external manifest", 2))
+		Expect(*external.Properties.AgentPoolProfiles[1].Name).To(Equal("compute"))
+		*external.Properties.AgentPoolProfiles[1].Count = 2
```

Comment:
You should be able to send a partial request to achieve the scale up. I am testing with a new manifest for my own purposes but you may want to reuse that. Will open a PR soon.

Reply:
Opened https://github.com/openshift/openshift-azure/pull/810

Reply:
sounds good. let me know when its out so I can have a look. In the meantime i'm trying to get us to use the client library for speaking to the RP (and also waiting for the geneva secrets blocker to be resolved)